be/src/vec/exec/format/table/hudi_jni_reader.cpp

```diff
@@ -65,16 +65,21 @@ HudiJniReader::HudiJniReader(const TFileScanRangeParams& scan_params,
             {"input_format", _hudi_params.input_format}};
 
     // Use compatible hadoop client to read data
-    for (auto& kv : _scan_params.properties) {
+    for (const auto& kv : _scan_params.properties) {
         if (kv.first.starts_with(HOODIE_CONF_PREFIX)) {
             params[kv.first] = kv.second;
         } else {
             params[HADOOP_CONF_PREFIX + kv.first] = kv.second;
         }
     }
 
-    _jni_connector = std::make_unique<JniConnector>("org/apache/doris/hudi/HudiJniScanner", params,
-                                                    required_fields);
+    if (config::use_old_hudi_jni_reader) [[unlikely]] {
```

Comment:
Better use a session variable

set hudi_jni_scanner = "spark" or "hadoop".
and default is hadoop